src/kv/kvtypes.h

```diff
@@ -191,6 +191,8 @@ namespace kv
     virtual void enable_all_domains() {}
     virtual void resume_replication() {}
     virtual void suspend_replication(kv::Version) {}
+
+    virtual void set_f(uint32_t f) = 0;
```

Comment:
It would be better to move the basic implementation from `raftconsensus` here so that `raftconsensus` doesn't need to worry about it, same as `on_request`.

Reply:
We don't want any implementation in kvtypes (or in anything *types in general), I agree it's probably good to share this logic, but not here.

Reply:
It's the same as `on_request`, we dummy implement it in the Consensus class so that when Raft extends this method there is no need for it to be overridden.

Reply:
Let's talk about this offline, it should be split out. ktypes.h should be a pure interface.